### ISSUE TITLE
Editing link blocks other options

### ISSUE THREAD

**Issue by DaNTaF** [WordPress/gutenberg#51430]
### Description

When I edit a link its blocks other options on the bar.
(RTL language)

1. Selected text with link (hover and block other options) : https://i.imgur.com/THMRDjn.png
2. Normal bar with all options: https://i.imgur.com/rmvaUPv.png

### Step-by-step reproduction instructions

(RTL language)

Select text and add a link
try to select this text again
the link menu blocks other options https://i.imgur.com/THMRDjn.png

Selected text without link works and does not block the options bar https://i.imgur.com/rmvaUPv.png

### Environment info

WordPress 6.2.2
RTL language

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by ndiego** [comment]
Hi @DaNTaF thanks for reporting. Does this happen all the time for you, or just based on the position of the text you are linking? In the gif below, you can see that I am able to replicate depending on the position of the text, but it doesn't happen all the time. 
![link-overlapping](https://github.com/WordPress/gutenberg/assets/4832319/66049794-a6d9-4da1-af46-9ead4625afb2)

**Comment by DaNTaF** [comment]
> Hi @DaNTaF thanks for reporting. Does this happen all the time for you, or just based on the position of the text you are linking? In the gif below, you can see that I am able to replicate depending on the position of the text, but it doesn't happen all the time. ![link-overlapping](https://user-images.githubusercontent.com/4832319/245527758-66049794-a6d9-4da1-af46-9ead4625afb2.gif)

Hey @ndiego  !

Please change your WordPress language to Hebrew and try again.
(or any other RTL language  )

**Comment by ndiego** [comment]
Yes, I am experiencing the same thing as my gif in RTL languages, both with Gutenberg and without Gutenberg enabled.

**Comment by DaNTaF** [comment]
@ndiego 
Maybe the problem is with short text

https://i.imgur.com/TE6OKRk.gif

**Comment by DaNTaF** [comment]
up
